### PR TITLE
add github workflow for performance benchmarking

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR addresses https://github.com/tensorflow/io/issues/1265 by adding a github workflow to benchmark API performance.
Benchmarking of additional API's will be taken up in the following PR's.

Additionally, I can add [github-action-benchmark](https://github.com/rhysd/github-action-benchmark/tree/master/examples/pytest) action to our workflow so that we can see the results in gh pages.
Please let me know.
cc: @yongtang @terrytangyuan

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          python --version
+          df -h
+          rm -rf tensorflow_io
+          echo ${{ matrix.version }} | awk -F: '{print $1}' | xargs python -m pip install -U
+          echo ${{ matrix.version }} | awk -F: '{print $2}' | xargs python -m pip install --no-deps -U
+          python -m pip install pytest-benchmark scikit-image
+          python -m pip freeze
+          python -c 'import tensorflow as tf; print(tf.version.VERSION)'
+          python -c 'import tensorflow_io as tfio; print(tfio.version.VERSION)'
+          python -m pytest --benchmark-only -v --import-mode=append $(find . -type f \( -iname "test_*_eager.py" ! \( -iname "test_bigquery_eager.py" \) \))
```
Comment (burgerkingeater) [inline .github/workflows/benchmarks.yml:42]: why is `test_bigquery_eager.py` handled differently?

Comment (yongtang) [reply .github/workflows/benchmarks.yml:42]: @burgerkingeater I remember that was due to https://github.com/grpc/grpc/issues/20034

**Review by yongtang — APPROVED** [review]

**Comment by yongtang** [comment]
> Additionally, I can add github-action-benchmark action to our workflow so that we can see the results in gh pages.

That would be great 👍

**Comment by yongtang** [comment]
The test is unrelated and will be fixed by #1270

**Comment by kvignesh1420** [comment]
I have added the `github-action-benchmark` step only for linux so that we get an idea of the look and feel in gh-pages.

**Review by yongtang — APPROVED** [review]
lgtm

**Pull request merged**